src/bbs-plus/encoder.ts

```diff
@@ -63,7 +63,13 @@ export class Encoder {
    * @param strict - If set to false and no appropriate encoder is found but the value is a bytearray, it will encode it using the built-in mechanism
    */
   encodeMessageObject(messages: object, strict = false): [string[], Uint8Array[]] {
-    const [names, values] = flattenObjectToKeyValuesList(messages);
+    let [names, values] = flattenObjectToKeyValuesList(messages);
+
+    // Filter keys to remove empty objects
+    // this is done because schema generation removes empty objects + nothing to encode
+    names = names.filter((k, i) => typeof values[i] !== 'object' || !isEmptyObject(values[i]));
```

Comment:
Similar comment as above, don't iterate here as the for loop below will do it. Move these checks inside it.

Reply:
thought it was  a bit cleaner to filter here rather than splicing in reverse loop then reversing the encoded array, but updated